lib/common/zstd_internal.h

```diff
@@ -188,8 +188,13 @@ static void ZSTD_copy16(void* dst, const void* src) {
     vst1q_u8((uint8_t*)dst, vld1q_u8((const uint8_t*)src));
 #elif defined(ZSTD_ARCH_X86_SSE2)
     _mm_storeu_si128((__m128i*)dst, _mm_loadu_si128((const __m128i*)src));
-#else
+#elif defined(__clang)
```

Comment:
This should be `__clang__` not `__clang`.

If x86-64 perf is not neutral, you could make this `elif defined(__clang__) || defined(__x86_64__) || defined(_M_X64)`.